examples/sandbox/index.ts

```diff
@@ -2518,139 +2508,132 @@ $ethAddr.on("click", async (e) => {
   } else {
     const label = await wallet.getLabel();
-    $ethResults.val(label + " does not support ETH");
+    ethButtons.ethResults.val(`${label} does not support ETH`);
   }
-});
-
-$ethSign.on("click", async (e) => {
-  e.preventDefault();
-  if (!wallet) {
-    $ethResults.val("No wallet?");
-    return;
-  }
-  if (core.supportsETH(wallet)) {
-    const { hardenedPath: hard, relPath: rel } = wallet.ethGetAccountPaths({
-      coin: "Ethereum",
-      accountIdx: 0,
-    })[0];
-    const result = await wallet.ethSignMessage({
-      addressNList: hard.concat(rel),
-      message: "Hello World",
-    });
-    $ethResults.val(result.address + ", " + result.signature);
-  } else {
-    const label = await wallet.getLabel();
-    $ethResults.val(label + " does not support ETH");
-  }
-});
-
-$ethSend.on("click", async (e) => {
-  e.preventDefault();
-  if (!wallet) {
-    $ethResults.val("No wallet?");
-    return;
-  }
-  if (core.supportsETH(wallet)) {
-    const result = ethEIP1559Selected
-      ? await wallet.ethSendTx(ethTx1559 as core.ETHSignTx)
-      : await wallet.ethSendTx(ethTx as core.ETHSignTx);
-    console.info("Result: ", result);
-    $ethResults.val(result.hash);
-  } else {
-    const label = await wallet.getLabel();
-    $ethResults.val(label + " does not support ETH");
-  }
-});
+};
 
-$ethVerify.on("click", async (e) => {
-  e.preventDefault();
-  if (!wallet) {
-    $ethResults.val("No wallet?");
-    return;
-  }
-  if (core.supportsETH(wallet)) {
-    const result = await wallet.ethVerifyMessage({
-      address: "0x2068dD92B6690255553141Dfcf00dF308281f763",
-      message: "Hello World",
-      signature:
-        "61f1dda82e9c3800e960894396c9ce8164fd1526fccb136c71b88442405f7d09721725629915d10bc7cecfca2818fe76bc5816ed96a1b0cebee9b03b052980131b",
-    });
-    $ethResults.val(result ? "✅" : "❌");
-  } else {
-    const label = await wallet.getLabel();
-    $ethResults.val(label + " does not support ETH");
-  }
-});
+ethButtons.ethAddr.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ hardenedPath, relPath }) => {
+      let result = await wallet.ethGetAddress({
+        addressNList: hardenedPath.concat(relPath),
+        showDisplay: false,
+      });
+      result = await wallet.ethGetAddress({
+        addressNList: hardenedPath.concat(relPath),
+        showDisplay: true,
+        address: result,
+      });
+      return result;
+    },
+  })
+);
+
+ethButtons.ethTx.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ tx }) => {
+      const res = await wallet.ethSignTx(tx);
+      return JSON.stringify(res);
+    },
+    useEIP1559: ethEIP1559Selected,
+  })
+);
+
+ethButtons.ethSign.on("click", (e) =>
+  performEthOperation(e, {
+    handler: async ({ hardenedPath, relPath }) => {
+      const result = await wallet.ethSignMessage({
+        addressNList: hardenedPath.concat(relPath),
+        message: "0x48656c6c6f20576f726c64", // "Hello World",
```

Comment:
This change is based on https://github.com/shapeshift/hdwallet/pull/635 that was missed in sandbox. Basically we should be passing a hex string here